terraform/api_gateway.tf.json.template.py

```diff
@@ -146,6 +147,70 @@ def check_waf_rules(rules: JSONs) -> JSONs:
 
 file_download_limit = config.waf_file_download_limit
 
+
+def add_waf_blocked_alarm(resources: JSON) -> JSON:
+    """
+    Adds a metric alarm based on WAF blocking rules
+    """
+    if config.enable_monitoring:
+        rules = [
+            rule['name']
+            for rule in resources['aws_wafv2_web_acl']['api_gateway']['rule']
+            if (
+                ('block' in rule.get('action', {}) or 'none' in rule.get('override_action', {}))
+                and rule['name'] not in config.waf_rules_not_logged
+            )
+        ]
+        metrics = [
+            ('AllowedRequests', 'ALL'),
+            *[('BlockedRequests', rule) for rule in rules]
+        ]
+        m_sum = '+'.join(f'm{i}' for i in range(1, len(metrics)))
+        expression = f'{m_sum}/(m0+{m_sum})*100'
+
+        assert 'aws_cloudwatch_metric_alarm' not in resources
+        resources['aws_cloudwatch_metric_alarm'] = {
```

Comment:
JSON isn't mutable. This will yield a warning once we cover it in mypy.

You can't declare a method JSON -> JSON and then have it modify the argument in place, and return it.